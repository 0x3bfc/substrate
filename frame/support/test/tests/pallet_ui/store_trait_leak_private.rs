#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::ModuleInterface;
	use frame_system::pallet_prelude::BlockNumberFor;
	use frame_support::pallet_prelude::StorageValueType;

	#[pallet::trait_]
	pub trait Trait: frame_system::Trait {}

	#[pallet::module]
	#[pallet::generate_store(pub trait Store)]
	pub struct Module<T>(core::marker::PhantomData<T>);

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface<BlockNumberFor<T>> for Module<T> {}

	#[pallet::call]
	impl<T: Trait> Module<T> {}

	#[pallet::storage]
	type Foo<T> = StorageValueType<_, u8>;
}

fn main() {
}
