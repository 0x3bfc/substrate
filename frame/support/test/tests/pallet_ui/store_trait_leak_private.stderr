error[E0446]: private type `pallet::_GeneratedPrefixForStorageFoo` in public interface
  --> $DIR/store_trait_leak_private.rs:11:37
   |
11 |     #[pallet::generate_store(pub trait Store)]
   |                                        ^^^^^ can't leak private type
...
21 |     type Foo = StorageValueType<_, u8>;
   |          - `pallet::_GeneratedPrefixForStorageFoo` declared as private
